Cython/Compiler/Nodes.py

```diff
@@ -140,6 +140,32 @@ def call(*args, **kwargs):
         return super().__new__(cls, name, bases, attrs)
 
 
+def CopyWithUpTreeRefsMixin(*uptree_refs_attrs):
+    class CopyWithUpTreeRefsMixin:
```

Comment:
You don't need to generate a class at runtime for this, just override a class attribute in the subclass to list the attribute names.

Reply:
BTW, `pos` (and probably some other attributes like the control flow state) can or should generally be excluded from deep-copying, so maybe we could have our own `__deepcopy__` implementation in the `Node` base class? We might even be able to find and exploit rules like "if the attribute refers to a `Node` but is not listed in `child_nodes`, then we could exclude it from the deepcopy by default" – something in that order.